system/src/ble_control_request_channel.cpp

```diff
@@ -659,6 +689,52 @@ void BleControlRequestChannel::run() {
     resetChannel();
 }
 
+int BleControlRequestChannel::setProvSvcUuid(hal_ble_uuid_t* svcUuid) {
+    if (!svcUuid) {
+        return SYSTEM_ERROR_INVALID_ARGUMENT;
+    }
+    bleCtrlSvcUuid_ = *svcUuid;
+    return SYSTEM_ERROR_NONE;
+}
+
+int BleControlRequestChannel::setProvTxUuid(hal_ble_uuid_t* txUuid) {
+    if (!txUuid) {
+        return SYSTEM_ERROR_INVALID_ARGUMENT;
+    }
+    bleSendCharUuid_ = *txUuid;
+    return SYSTEM_ERROR_NONE;
+}
+
+int BleControlRequestChannel::setProvRxUuid(hal_ble_uuid_t* rxUuid) {
+    if (!rxUuid) {
+        return SYSTEM_ERROR_INVALID_ARGUMENT;
+    }
+    bleRecvCharUuid_ = *rxUuid;
+    return SYSTEM_ERROR_NONE;
+}
+
+int BleControlRequestChannel::setProvVerUuid(hal_ble_uuid_t* verUuid) {
+    if (!verUuid) {
+        return SYSTEM_ERROR_INVALID_ARGUMENT;
+    }
+    bleVerCharUuid_ = *verUuid;
+    return SYSTEM_ERROR_NONE;
+}
+
+hal_ble_uuid_t BleControlRequestChannel::getBleCtrlSvcUuid() {
+    return bleCtrlSvcUuid_;
+}
+
+int BleControlRequestChannel::setCompanyId(uint16_t companyId) {
+    // TODO: Add error checking for companyId
```

Comment:
This API should belong to the `BleProvisioningModeHandler` class, as the BLE control request doesn't care what the company ID is, it is used in advertising data only. We probably don't need to add any check here. It's all up to who invokes this API.